test/e2e/util.go

```diff
@@ -517,32 +517,53 @@ func waitForRCPodToDisappear(c *client.Client, ns, rcName, podName string) error
 	})
 }
 
-// waits until the service appears (exists == true), or disappears (exists == false)
-func waitForService(c *client.Client, namespace, name string, exist bool, interval, timeout time.Duration) error {
-	return wait.Poll(interval, timeout, func() (bool, error) {
+// waitForService waits until the service appears (toAppear == true), or disappears (toAppear == false)
+func waitForService(c *client.Client, namespace, name string, toAppear bool, interval, timeout time.Duration) error {
+	err := wait.Poll(interval, timeout, func() (bool, error) {
 		_, err := c.Services(namespace).Get(name)
 		if err != nil {
 			Logf("Get service %s in namespace %s failed (%v).", name, namespace, err)
-			return !exist, nil
+			return !toAppear, nil
 		} else {
 			Logf("Service %s in namespace %s found.", name, namespace)
-			return exist, nil
+			return toAppear, nil
 		}
 	})
+	if err != nil {
+		var desiredStatus string
+		if toAppear {
+			desiredStatus = " to appear"
+		} else {
+			desiredStatus = " to dissappear"
```

Comment:
tiny tiny nit: if you change anything else, can you take the space out of this and put it in the Errorf below? I think it'd be a tiny bit easier to read...